File: ScheduleDemoApp.Web/Models/ViewModels/CommitmentModel.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace ScheduleDemoApp.Models.ViewModels
{
    public class CommitmentModel
    {
        public int id { get; set; }
        public string location { get; set; }
        public string subject { get; set; }
        public string body { get; set; }
        public DateTime startDate { get; set; }
        public DateTime endDate { get; set; }
        public CategoryModel category { get; set; }
        public IEnumerable<CommitmentPersonModel> commitmentPeople { get; set; }
    }
}

File: ScheduleDemoApp.Web/ClientApp/app/models/commitment.model.ts
import { Person } from './person.model';
import { Category } from './category.model';


export class Commitment {
    id: number;
    location: string;
    subject: string;
    body: string;
    startDate: Date;
    endDate: Date;
    category: Category;
    people: Person[];
}
